packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -284,13 +276,37 @@ export class SerializedStateManager {
 		}
 	}
 
+	private tryRefreshSnapshot(): void {
+		if (
+			this._refreshSnapshotP === undefined &&
+			this.mc.config.getBoolean("Fluid.Container.enableOfflineSnapshotRefresh") === true
+		) {
+			// Don't block on the refresh snapshot call - it is for the next time we serialize, not booting this incarnation
+			this._refreshSnapshotP = this.refreshLatestSnapshot(this.supportGetSnapshotApi());
+			this._refreshSnapshotP
+				.catch(
+					(error: TelemetryEventPropertyTypeExt | Tagged<TelemetryEventPropertyTypeExt>) => {
+						this.mc.logger.sendTelemetryEvent({
+							eventName: "RefreshLatestSnapshotFailed",
+							error,
+						});
+						this._refreshSnapshotP = undefined;
+						this.tryRefreshSnapshot();
```

Comment:
is there anything preventing this from overlapping with the timer? do we do anything make sure only 1 tryRefreshSnapshot is ever happening at once?

Reply:
I believe by removing this tryRefreshSnapshot and blocking when latestSnapshot is set should be enough to prevent overlapping but I can wrap this function on a runSingle

Reply:
I believe the runSingle approach doesn't work since tryRefreshSnapshot is not async. Conditions should be sufficient to prevent reentrancy.